app/bundles/ApiBundle/Form/Type/ClientType.php

```diff
@@ -44,19 +48,33 @@ class ClientType extends AbstractType
     private $apiMode;
 
     /**
-     * @var \Symfony\Bundle\FrameworkBundle\Routing\Router
+     * @var \Symfony\Component\Routing\RouterInterface
      */
     private $router;
 
     /**
-     * @param MauticFactory $factory
+     * Constructor.
+     *
+     * @param ServiceContainerInterface $container
+     * @param TranslatorInterface       $translator
+     * @param ValidatorInterface        $validator
+     * @param Session                   $session
+     * @param RouterInterface           $router
      */
-    public function __construct(MauticFactory $factory)
-    {
-        $this->translator = $factory->getTranslator();
-        $this->validator  = $factory->getValidator();
-        $this->apiMode    = $factory->getRequest()->get('api_mode', $factory->getSession()->get('mautic.client.filter.api_mode', 'oauth1a'));
-        $this->router     = $factory->getRouter();
+    public function __construct(
+        ContainerInterface $container,
+        TranslatorInterface $translator,
+        ValidatorInterface $validator,
+        Session $session,
+        RouterInterface $router
+    ) {
+        $this->translator = $translator;
+        $this->validator  = $validator;
+        $this->apiMode    = $container->get('request')->get(
```

Comment:
Use RequestStack instead of the ContainerInterface if it's used only to get the current Request.